zlib issue on RHEL

When trying to build all contrib packages I get the following error

```
$ cmake -DBUILD_TYPE=ALL .
[...]
-- Downloading ZLIB ..
-- Downloading ZLIB .. done
-- Extracting ZLIB ..
-- Extracting ZLIB .. done
-- Configuring zlib library (./configure --prefix=/scratch/hroest/openms/test/test_new_oMS2/contrib  CC=/usr/bin/cc) ..
-- Configuring zlib library (./configure --prefix=/scratch/hroest/openms/test/test_new_oMS2/contrib  CC=/usr/bin/cc) .. done
-- Building zlib library (make CC='/usr/bin/cc' CFLAGS='-Wall -O3 -fPIC') ..
-- Building zlib library (make CC='/usr/bin/cc' CFLAGS='-Wall -O3 -fPIC') .. done
-- Installing zlib library (make install) ..
-- Installing zlib library (make install) .. failed
CMake Error at libraries.cmake/zlib.cmake:146 (message):
  cp libz.a /scratch/hroest/openms/test/test_new_oMS2/contrib/lib

  cp /scratch/hroest/openms/test/test_new_oMS2/contrib/lib

  cp: missing destination file operand after
  `/scratch/hroest/openms/test/test_new_oMS2/contrib/lib'

  Try `cp --help' for more information.

  gmake: *** [install-libs] Error 1
Call Stack (most recent call first):
  CMakeLists.txt:418 (OPENMS_CONTRIB_BUILD_ZLIB)


-- Configuring incomplete, errors occurred!

```

I printed out the output of configure using
```
--- a/libraries.cmake/zlib.cmake
+++ b/libraries.cmake/zlib.cmake
@@ -107,8 +107,23 @@ MACRO( OPENMS_CONTRIB_BUILD_ZLIB )
       message( FATAL_ERROR ${ZLIB_CONFIGURE_OUT})
     else()
       message( STATUS "Configuring zlib library (./configure --prefix=${CMAKE_BINARY_DIR} ${ZLIB_EXTRA_FLAGS} CC=${CMAKE_C_COMPILER}) .. done")
+      message( STATUS ${ZLIB_CONFIGURE_OUT})
     endif()
```

and oddly enough I get two different outputs when running it twice:
```
Tested /usr/bin/cc -w -c -O ztest160411.c
Tested cc -shared -O -o ztest160411.so ztest160411.o
/usr/bin/ld: ztest160411.o: relocation R_X86_64_PC32 against undefined symbol `getchar@@GLIBC_2.2.5' can not be used when making a shared object recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: ld returned 1 exit status
No shared library support try without defining CC and CFLAGS
Building static library libz.a version 1.2.5 with /usr/bin/cc.
Checking for off64_t... Yes.
Checking for fseeko... Yes.
Checking for unistd.h... Yes.
Checking whether to use vs[n]printf() or s[n]printf()... using vs[n]printf().
Checking for vsnprintf() in stdio.h... Yes.
Checking for return value of vsnprintf()... Yes.
```

and second time:
```
-- Checking for gcc...
Checking for shared library support...
Building shared library libz.so.1.2.5 with gcc.
Checking for off64_t... Yes.
Checking for fseeko... Yes.
Checking for unistd.h... Yes.
Checking whether to use vs[n]printf() or s[n]printf()... using vs[n]printf().
Checking for vsnprintf() in stdio.h... Yes.
Checking for return value of vsnprintf()... Yes.
Checking for attribute(visibility) support... Yes.
```

--- a/libraries.cmake/zlib.cmake
+++ b/libraries.cmake/zlib.cmake
@@ -80,6 +80,14 @@ MACRO( OPENMS_CONTRIB_BUILD_ZLIB )
   
 	else() ## Linux/MacOS  
 
+    # Workaround for bug https://github.com/OpenMS/contrib/issues/5 
+    # Basically, the configure script of zlib has a nasty property of checking
+    # whether the current compiler is gcc or not by direct string comparison.
+    # Apparently it really likes gcc and if the $CC variable is set to anything
+    # else (or even set at all), it will likely refuse to work.
+    set(old_CC $ENV{CC})
+    UNSET(ENV{CC})
+
 	  # CFLAGS for libsvm compiler (see libsvm Makefile)
     set(ZLIB_CFLAGS "-Wall -O3 -fPIC")
 
@@ -147,6 +155,9 @@ MACRO( OPENMS_CONTRIB_BUILD_ZLIB )
     else()
       message( STATUS "Installing zlib library (make install) .. done")
     endif()
+
+    # reset CC, see https://github.com/OpenMS/contrib/issues/5 
+    SET(ENV{CC} ${old_CC})
 
 endif()
 

--- a/libraries.cmake/zlib.cmake
+++ b/libraries.cmake/zlib.cmake
@@ -97,12 +97,11 @@ MACRO( OPENMS_CONTRIB_BUILD_ZLIB )
     endif( ${CMAKE_SYSTEM_NAME} MATCHES "Darwin" )
 
 		# configure with with prefix
-    message( STATUS "Configuring zlib library (./configure --prefix=${CMAKE_BINARY_DIR} ${ZLIB_EXTRA_FLAGS} CC=${CMAKE_C_COMPILER}) .. ")
+    message( STATUS "Configuring zlib library (./configure --prefix=${CMAKE_BINARY_DIR} ${ZLIB_EXTRA_FLAGS}) .. ")
     exec_program("./configure" ${ZLIB_DIR}
       ARGS
       --prefix=${CMAKE_BINARY_DIR}
       ${ZLIB_EXTRA_FLAGS}
-      #CC=${CMAKE_C_COMPILER}
       OUTPUT_VARIABLE ZLIB_CONFIGURE_OUT
       RETURN_VALUE ZLIB_CONFIGURE_SUCCESS
       )  
@@ -111,19 +110,20 @@ MACRO( OPENMS_CONTRIB_BUILD_ZLIB )
     file(APPEND ${LOGFILE} ${ZLIB_CONFIGURE_OUT})
 
     if( NOT ZLIB_CONFIGURE_SUCCESS EQUAL 0)
-      message( STATUS "Configuring zlib library (./configure --prefix=${CMAKE_BINARY_DIR} ${ZLIB_EXTRA_FLAGS} CC=${CMAKE_C_COMPILER}) .. failed")
+      message( STATUS "Configuring zlib library (./configure --prefix=${CMAKE_BINARY_DIR} ${ZLIB_EXTRA_FLAGS}) .. failed")
       message( FATAL_ERROR ${ZLIB_CONFIGURE_OUT})
     else()
-      message( STATUS "Configuring zlib library (./configure --prefix=${CMAKE_BINARY_DIR} ${ZLIB_EXTRA_FLAGS} CC=${CMAKE_C_COMPILER}) .. done")
+      message( STATUS "Configuring zlib library (./configure --prefix=${CMAKE_BINARY_DIR} ${ZLIB_EXTRA_FLAGS}) .. done")
     endif()
 		
 
 		# make
-    message(STATUS "Building zlib library (make CC='${CMAKE_C_COMPILER}' CFLAGS='${ZLIB_CFLAGS}') .. ")
+    message(STATUS "Building zlib library (make CFLAGS='${ZLIB_CFLAGS}') .. ")
     exec_program(${CMAKE_MAKE_PROGRAM} ${ZLIB_DIR}
       ARGS # setting compiler and flags
       clean all
-      CC=${CMAKE_C_COMPILER}
+      # We let ./configure choose the compiler, lets not interfere
+      # CC=${CMAKE_C_COMPILER}
       CFLAGS='${ZLIB_CFLAGS}'
       OUTPUT_VARIABLE ZLIB_MAKE_OUT
       RETURN_VALUE BUILD_SUCCESS)
@@ -132,10 +132,10 @@ MACRO( OPENMS_CONTRIB_BUILD_ZLIB )
     file(APPEND ${LOGFILE} ${ZLIB_MAKE_OUT})
 
 	  if (NOT BUILD_SUCCESS EQUAL 0)
-      message(STATUS "Building zlib library (make CC='${CMAKE_C_COMPILER}' CFLAGS='${ZLIB_CFLAGS}') .. done")
+      message(STATUS "Building zlib library (make CFLAGS='${ZLIB_CFLAGS}') .. done")
 	    Message(FATAL_ERROR ${ZLIB_MAKE_OUT})
     else()
-      message(STATUS "Building zlib library (make CC='${CMAKE_C_COMPILER}' CFLAGS='${ZLIB_CFLAGS}') .. done")
+      message(STATUS "Building zlib library (make CFLAGS='${ZLIB_CFLAGS}') .. done")
 	  endif()
     
     # make install

